lib/internal/process.js

```diff
@@ -78,14 +78,19 @@ function setup_hrtime() {
     _hrtime(hrValues);
 
     if (time !== undefined) {
-      if (Array.isArray(time) && time.length === 2) {
-        const sec = (hrValues[0] * 0x100000000 + hrValues[1]) - time[0];
-        const nsec = hrValues[2] - time[1];
-        const needsBorrow = nsec < 0;
-        return [needsBorrow ? sec - 1 : sec, needsBorrow ? nsec + 1e9 : nsec];
+      if (!Array.isArray(time)) {
+        throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'time', 'Array',
+                                   time);
       }
-      throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
-                                 'process.hrtime()', 'Array');
+      if (time.length !== 2) {
+        throw new errors.TypeError('ERR_INVALID_ARRAY_LENGTH', 'time', 2,
```

Comment:
RangeError is [defined in ES](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-rangeerror) as 

> Indicates a value that is not in the set or range of allowable values.

Here, it's not that *time* is not in the set or range of allowable values, but rather a property of it, so TypeError should be more appropriate.

The definition also implies that RangeError can also be used for strings, such as in [String.prototype.normalize](https://tc39.github.io/ecma262/#sec-string.prototype.normalize).